Example_Programs/programming_concpets/ttc_1a/inc/header_def.h

```diff
@@ -0,0 +1,56 @@
+/**
+ * @file header_def.h
+ * @author Arnob
+ * @brief 
+ * @version 0.1
+ * @date 2021-03-21
+ * 
+ * @copyright Copyright (c) 2021
+ * 
+ */
+
+#ifndef __FUN_H__
+#define __FUN_H___
+
+/**
+ * @brief User defined datatype to hold imaginary and real part of imaginary number
+ * 
+ */
+typedef struct complex_num
+{
+    int real, img;
+} complex_num;
+/**
+ * @brief add two complex numbers
+ * 
+ * @param c_1 
+ * @param c_2 
+ * @return Integer
+ */
+int sum(complex_num c_1, complex_num c_2);
```

Comment:
Not a portable solution. If the user want to reuse and do some operation on the calculated result, its not possible with this implementation.

Reply:
This code can be considered deprecated as I completed and pushed the template complex calculator repository where I used pass by reference(emulated).